docs/topics/upgrade.md

```diff
@@ -0,0 +1,77 @@
+# Upgrading Kubernetes Clusters
+
+## Prerequisites
+
+All the commands in this guide require both the Azure CLI and `aks-engine`. Follow the [quickstart guide](../tutorials/quickstart.md) before continuing.
+
+This guide assumes you already have deployed a cluster using `aks-engine`. For more details on how to do that see [deploy](../tutorials/deploy.md).
+
+## Upgrade
+
+This document provides guidance on how to upgrade the Kubernetes version for an existing `aks-engine` cluster and recommendations for adopting `aks-engine upgrade` as a production quality tool.
+
+<a name="pre-requirements"></a>
+
+### Know before you go
+
+In order to ensure that your `aks-engine upgrade` operation runs smoothly, there are a few things you should be aware of before getting started.
+
+1) You will need access to the original `_output/` directory files that were generated during cluster deployment in order to run `upgrade`.  `aks-engine` will use the `--deployment-dir` argument provided in the upgrade command to introspect the Azure Resource Manager (ARM) template and parameters to determine the cluster's current Kubernetes version and its current state.
+
+2) `aks-engine upgrade` expects a cluster configuration that conforms to the current state of the cluster. In other words, the Azure resources deployed by `aks-engine` should be in the same state that when they were originally created by `aks-engine`. If you perform manual operations on your Azure IaaS resources (other than `aks-engine scale` and `aks-engine upgrade`) you risk not being able to upgrade your cluster successfully as `aks-engine` won't be able to reconcile the cluster's current state. This includes naming of resources; `aks-engine` relies on some resources (such as VMs) to be named in accordance with the original `aks-engine` deployment.
+
+3) `aks-engine upgrade` allows one subsequent minor version upgrade at a time, as long as the next minor version is supported. If the next minor version is deprecated, `aks-engine` allows upgrade to the next supported minor version.
+
+To get the list of all available Kubernetes versions and upgrades, run the `orchestrators` command and specify Kubernetes orchestrator type:
+
+```bash
+./bin/aks-engine orchestrators --orchestrator Kubernetes
+```
+
+To get the information specific to the cluster, provide its current Kubernetes version in the `version` arg:
```

Comment:
s/To get the information specific to the cluster/To discover the versions of Kubernetes that your particular cluster version is upgradable to/